src/loaders/genome_attributes/parse_computed_genome_attribs.py

```diff
@@ -0,0 +1,247 @@
+    result_dir = os.path.join(root_dir, loader_common_names.COLLECTION_DATA_DIR, kbase_collection, load_ver, tool)
+
+    if check_exists and not (os.path.exists(result_dir) and os.path.isdir(result_dir)):
+        raise ValueError(f"Result directory for computed genome attributes of "
+                         f"Kbase Collection: {kbase_collection} and Load Version: {load_ver} could not be found.")
+
+    return result_dir
+
+
+def _read_tsv_as_df(file_path, features, genome_id_col=None):
+    # Retrieve the desired fields from a TSV file and return the data in a dataframe
+
+    selected_cols = copy.deepcopy(features) if features else None
+
+    if selected_cols and genome_id_col:
+        selected_cols.add(genome_id_col)
+
+    df = pd.read_csv(file_path, sep='\t', keep_default_na=False, usecols=selected_cols)
+
+    return df
+
+
+def _create_doc(row, kbase_collection, load_version, genome_id, features, pre_fix):
+    # Select specific columns and prepare them for import into Arango
+
+    # NOTE: The selected column names will have a prefix added to them if pre_fix is not empty.
+
+    doc = init_genome_atrri_doc(kbase_collection, load_version, genome_id)
+
+    # distinguish the selected fields from the original metadata by adding a common prefix to their names
+    if features:
+        doc.update(row[list(features)].rename(lambda x: pre_fix + '_' + x if pre_fix else x).to_dict())
+    else:
+        doc.update(row.rename(lambda x: pre_fix + '_' + x if pre_fix else x).to_dict())
+
+    return doc
+
+
+def _row_to_doc(row, kbase_collection, load_version, features, tool_genome_map, genome_id_col, pre_fix):
+    # Transforms a row from tool result file into ArangoDB collection document
+
+    try:
+        genome_id = tool_genome_map[row[genome_id_col]]
+    except KeyError as e:
+        raise ValueError('Unable to find genome ID') from e
+
+    doc = _create_doc(row, kbase_collection, load_version, genome_id, features, pre_fix=pre_fix)
+
+    return doc
+
+
+def _read_tool_result(result_dir, batch_dir, kbase_collection, load_ver, tool_file_name, features, genome_id_col,
+                      pre_fix=''):
+    # process the output file generated by the tool (checkm2, gtdb-tk, etc) to create a format suitable for importing
+    # into ArangoDB
+    batch_result_dir = os.path.join(result_dir, str(batch_dir))
+
+    # retrieve and process the genome metadata file
+    metadata_file = os.path.join(batch_result_dir, loader_common_names.GENOME_METADATA_FILE)
+    try:
+        meta_df = pd.read_csv(metadata_file, sep='\t', names=['tool_identifier', 'genome_id', 'source_file_path'])
+    except Exception as e:
+        raise ValueError('Unable to retrieve the genome metadata file') from e
+    tool_genome_map = dict(zip(meta_df.tool_identifier, meta_df.genome_id))
+
+    tool_file = os.path.join(result_dir, str(batch_dir), tool_file_name)
+    docs = dict()
+    if os.path.exists(tool_file):
```

Comment:
If it doesn't exist isn't that an error?

Reply:
gotcha. can you document that the method returns an empty dict if the file is missing?

Reply:
👍

Reply:
:+1: